x-pack/legacy/plugins/monitoring/server/lib/logstash/get_paginated_pipelines.js

```diff
@@ -45,61 +46,155 @@ export async function getPaginatedPipelines(
     size
   );
 
-  // `metricSet` defines a list of metrics that are sortable in the UI
-  // but we don't need to fetch all the data for these metrics to perform
-  // the necessary sort - we only need the last bucket of data so we
-  // fetch the last two buckets of data (to ensure we have a single full bucekt),
-  // then return the value from that last bucket
+  if (sortField === throughputMetric) {
```

Comment:
I don't really know if we need these changes. I can follow the code and they make sense, but it leads to duplication in the `get...Data` functions themselves. The thinking behind not separating the calls originally is the calls should be super efficient, as the date histogram will only capture a couple full buckets of data (instead of the entire time range). What did you see here that led you to want to make these changes?

Reply:
This was actually another big optimization, since before this we were fetching *all* `nodesCountMetric` pipelines per every `throughputMetric` pipeline. So, if you had 10 pipelines we would make 10*10 calls (instead of 10+1) to ES. This was because `throughputMetric` calls are indeed one per pipeline, where nodesCountMetric are a single call per `pageOfPipelines` however we were passing `pipeline` (object) instead of `pageOfPipelines` (array), which as a result got all the pipelines (since it was done for both metrics in [Array.map](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/monitoring/server/lib/logstash/get_paginated_pipelines.js#L55-L60))

So, while fixing the issue mentioned above, I found it unnecessary to also fetch `nodesCountMetric` if we are only sorting by `throughputMetric` (or vise-versa), since these are separate metric calls

Reply:
Ah okay, I see that now! Yes, that looks like an oversight and a legitimate bug, nice find!